src/presenter/films-presenter.js

```diff
@@ -78,58 +71,53 @@ export default class FilmsPresenter {
     this.#renderTopAndMost();
   }
 
-  #renderFilm(filmModelCard, filmContainer) {
+  #handleShowMoreButtonClick = () => {
+    this.#filmsModelContainer
+      .slice(this.#renderedFilmModelCard, this.#renderedFilmModelCard + FILM_CARD_PER_STEP)
+      .forEach((filmModelCard) => this.#renderFilm(filmModelCard, this.#filmListComponent.element));
 
-    const escKeyDownPopUp = (evt) => {
-      if (evt.key === 'Escape' || evt.key === 'Esc') {
-        evt.preventDefault();
-        closePopUp();
-        document.removeEventListener('keydown', escKeyDownPopUp);
-      }
-    };
-
-    const filmCardComponent = new FilmCardView({
-      filmModelCard,
-      onFilmCardClick: () => {
-        openPopUp();
-        document.addEventListener('keydown', escKeyDownPopUp);
-      }
-    });
+    this.#renderedFilmModelCard += FILM_CARD_PER_STEP;
 
-    const infoPopUpComponent = new InfoPopUpView({
-      filmModelCard,
-      onPopUpClick: () => {
-        closePopUp();
-        document.removeEventListener('keydown', escKeyDownPopUp);
-      }
-    });
-    function openPopUp () {
-      bodyElement.appendChild(infoPopUpComponent.element);
-      bodyElement.classList.add('.hide-overflow');
+    if (this.#renderedFilmModelCard >= this.#filmsModelContainer.length) {
+      remove(this.#showMoreButtonComponent);
     }
+  };
 
-    function closePopUp () {
-      bodyElement.removeChild(infoPopUpComponent.element);
-      bodyElement.classList.remove('.hide-overflow');
-    }
+  #handleFilmCardChange = (updatedFilm) => {
+    this.#filmsModelContainer = updateFilmCard(this.#filmsModelContainer, updatedFilm);
+    this.#cardFilmPresenter.get(updatedFilm.id).init(updatedFilm);
+  };
+
+
+  #renderFilm(filmModelCard, filmContainer) {
+    const cardFilmPresenter = new CardFilmPresenter({
+      filmContainerTeg: filmContainer,
+      onDataChange: this.#handleFilmCardChange
+    });
 
-    render(filmCardComponent, filmContainer);
+    cardFilmPresenter.init(filmModelCard);
+    this.#cardFilmPresenter.set(filmModelCard.id, cardFilmPresenter);
   }
 
+
   #renderTopAndMost(){
 
     render(this.#topRatedList, this.#filmComponent.element);
     render(this.#topRatedContainer, this.#topRatedList.element);
 
+    this.#mostRatedFilms = topRatedFilmSort(this.#filmsModelContainer);
+    this.#mostCommentedFilms = commentedFilmSort(this.#filmsModelContainer);
```

Comment:
Конечно `sort` является глаголом, но эти функции ведь не только сортируют, они ещё обрезают до нужно длинны, поэтому я бы назвал так:
```suggestion
    this.#mostRatedFilms = getTopRatedFilms(this.#filmsModelContainer);
    this.#mostCommentedFilms = getMostCommentedFilms(this.#filmsModelContainer);
```